### ISSUE TITLE
Specify preloadURL / preload for a tab

### ISSUE THREAD

**Issue by gknapp** [brrd/electron-tabs#72]
As each tab has it's own `webview`, I wanted to know if I could specify a preload script?

I'd like to set a variable on the `window` object. My web application (loaded in a tab) can detect the variable and behave accordingly.

Example: https://medium.com/cameron-nokes/how-to-create-a-hybrid-electron-app-53553ece0889

I've tried specifying a `preload` script in `webviewAttributes` in tab options when adding a tab to the tab group, the script never runs though.

Is there another way to specify a preload script for a tab?

I'm using electron 1.8.8 and electron-tabs 0.9.4.

**Comment by brrd** [comment]
> I've tried specifying a preload script in webviewAttributes in tab options when adding a tab to the tab group, the script never runs though.

This should work normally:

```javascript
tabGroup.addTab({
  webviewAttributes: {
    preload: "./test.js"
  }
});
```
> Is there another way to specify a preload script for a tab?

No. `webviewAttributes` simply adds attributes to the webview element, so it is supposed to be the most efficient way to add a preload script.

**Comment by gknapp** [comment]
Thanks for clarifying. I'll test it again tomorrow.

**Comment by gknapp** [comment]
So I managed to get a console to display by doing the following (when I looked closely, I'd done this elsewhere in my code base):

```js
const tab = group.addTab({
  webviewAttributes: {
    preload: config.preload
  }
});
tab.on(
  "webview-ready",
  t => t.webview.openDevTools()
);
```

That helped a lot in debugging the issue. I couldn't use an absolute path (as I had use elsewhere in the app).

As you suggested `"./preload.js"` worked in the end. It didn't help I had an `import` statement in `preload.js` that caused the module to fail, even when `preload.js` was loaded successfully (I had not included the imported file in my `build` directory).
